src/EnergyPlus/HVACManager.cc

```diff
@@ -2995,4 +2904,46 @@ void CheckAirLoopFlowBalance(EnergyPlusData &state)
     }
 }
 
+void ConvergenceErrors(EnergyPlusData &state,
+                       std::array<bool, 3> HVACNotConverged,
+                       std::array<Real64, 10> DemandToSupply,
+                       std::array<Real64, 10> SupplyDeck1ToDemand,
+                       std::array<Real64, 10> SupplyDeck2ToDemand,
+                       int const AirSysNum,
+                       int const index)
+{
+    std::string CaseName;
+    if (index == 1) CaseName = "mass flow rate";
+    if (index == 2) CaseName = "humidity ratio";
+    if (index == 3) CaseName = "temperature";
+    if (index == 4) CaseName = "energy";
+    if (index == 5) CaseName = "CO2";
+    if (index == 6) CaseName = "generic contaminant";
```

Comment:
Oooh this should be a constexpr array of strings.  I'm going to stop short of requiring it here, but this should just be a lookup.